test/Altinn.App.Tests.Common/Utils/OtelVisualizers.cs

```diff
@@ -7,12 +7,22 @@ namespace Altinn.App.Tests.Common.Utils;
 public static class OtelVisualizers
 {
     public static string VisualizeActivities(
-        List<Activity> activities,
-        List<LogRecord> logs,
+        List<Activity> activitiesInput,
+        List<LogRecord> logsInput,
         bool includeDuration = true,
         int initialIndent = 0
     )
     {
+        // This is a little racy potentially,
+        // got some enumeration errors that hopefully is avoided by taking a copy of the input first
+        // The inputs can only grow, so worst case we miss a few elements if the race hits
```

Comment:
I think the issue here might be caused by not disposing the service provider or flushing logs manually before printing the logs.

https://github.com/open-telemetry/opentelemetry-dotnet/issues/5102

I should probably just follow the instructions and dispose the _serviceProvider before visualising telemetry, instead of trying to ForceFlush.


The issue was that the API gave me no LoggerProvider to call `ForceFlush` on.
var loggerProvider = app.Services.GetRequiredService<LoggerProvider>();